tools/lsp/preview/drop_location.rs

```diff
@@ -99,19 +108,34 @@ fn find_drop_location(
         result
     }?;
 
-    let insertion_position = target_element_node.with_element_node(|node| {
-        let last_token = crate::util::last_non_ws_token(node)?;
+    let (insertion_position, indent, replacement_range) = target_element_node.with_element_node(|node| {
+        let closing_brace = crate::util::last_non_ws_token(node)?;
+        let closing_brace_offset = Into::<u32>::into(closing_brace.text_range().start());
+
+        let before_closing = closing_brace.prev_token().unwrap(); // there must be something before the closing bracket
```

Comment:
in general, it is better to have comment on the line before, because otherwise, auto-formatter might mess that up and put it in the next line.
In this case, the comment could be in a expect("...")
but why not use `?` ?